entity-framework/core/miscellaneous/spatial.md

```diff
@@ -0,0 +1,295 @@
+
+## Installing
+
+In order to use spatial data with EF Core, you need to install the appropriate supporting NuGet package. Which package you need to install depends on the provider you're using.
+
+EF Core Provider                        | Spatial NuGet Package
+--------------------------------------- | ---------------------
+Microsoft.EntityFrameworkCore.SqlServer | [Microsoft.EntityFrameworkCore.SqlServer.NetTopologySuite](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.SqlServer.NetTopologySuite)
+Microsoft.EntityFrameworkCore.Sqlite    | [Microsoft.EntityFrameworkCore.Sqlite.NetTopologySuite](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Sqlite.NetTopologySuite)
+Microsoft.EntityFrameworkCore.InMemory  | [NetTopologySuite](https://www.nuget.org/packages/NetTopologySuite)
+Npgsql.EntityFrameworkCore.PostgreSQL   | [Npgsql.EntityFrameworkCore.PostgreSQL.NetTopologySuite](https://www.nuget.org/packages/Npgsql.EntityFrameworkCore.PostgreSQL.NetTopologySuite)
+
+## Reverse engineering
+
+The spatial NuGet packages also enable [reverse engineering](../managing-schemas/scaffolding.md) models with spatial properties, but you need to install the package ***before*** running `Scaffold-DbContext` or `dotnet ef dbcontext scaffold`. If you don't, you'll receive warnings about not finding type mappings for the columns and the columns will be skipped.
+
+## NetTopologySuite (NTS)
+
+NetTopologySuite is a spatial library for .NET. EF Core enables mapping to spatial data types in the database by using NTS types in your model.
+
+To enable mapping to spatial types via NTS, call the UseNetTopologySuite method on the provider's DbContext options builder. For example, with SQL Server you'd call it like this.
+
+``` csharp
+optionsBuilder.UseSqlServer(
+    @"Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=WideWorldImporters",
+    x => x.UseNetTopologySuite());
+```
+
+There are several spatial data types. Which type you use depends on the types of shapes you want to allow. Here is the hierarchy of NTS types that you can use for properties in your model. They're located within the `NetTopologySuite.Geometries` namespace. Corresponding interfaces in the GeoAPI package can also be used.
+
+* Geometry
+  * Point
+  * LineString
+  * Polygon
+  * GeometryCollection
+    * MultiPoint
+    * MultiLineString
+    * MultiPolygon
+
+> [!WARNING]
+> CircularString, CompoundCurve, and CurePolygon aren't supported by NTS.
+
+The following entity classes could be used to map to tables in the [Wide World Importers sample database](http://go.microsoft.com/fwlink/?LinkID=800630).
+
+``` csharp
```

Comment:
Are you planning to create a working sample project? I think in this case it could be very useful, but no problem with doing it afterwards.

Reply:
We should consider moving [this sample](https://github.com/divega/EFDemos/tree/master/spatial) into either here or the main repo.